provisioner/terraform/modules.go

```diff
@@ -62,3 +70,54 @@ func getModules(workdir string) ([]*proto.Module, error) {
 	}
 	return filteredModules, nil
 }
+
+func getModulesArchive(workdir string) ([]byte, error) {
+	empty := true
+	var b bytes.Buffer
+	w := tar.NewWriter(&b)
+	modulesDir := getModulesDirectory(workdir)
+	err := filepath.WalkDir(modulesDir, func(filePath string, info fs.DirEntry, err error) error {
+		if err != nil {
+			return xerrors.Errorf("failed to create modules archive: %w", err)
+		}
+		if info.IsDir() {
+			return nil
+		}
+		empty = false
+		archivePath, found := strings.CutPrefix(filePath, workdir+"/")
+		if !found {
+			return xerrors.Errorf("walked invalid file path: %q", filePath)
+		}
+
+		content, err := os.ReadFile(filePath)
+		if err != nil {
+			return xerrors.Errorf("failed to read module file while archiving: %w", err)
+		}
+		err = w.WriteHeader(&tar.Header{
+			Name: archivePath,
+			Size: int64(len(content)),
+			Mode: 0o644,
+			Uid:  1000,
+			Gid:  1000,
+		})
+		if err != nil {
+			return xerrors.Errorf("failed to add module file to archive: %w", err)
+		}
+		if _, err = w.Write(content); err != nil {
+			return xerrors.Errorf("failed to write module file to archive: %w", err)
+		}
+		return nil
+	})
```

Comment:
We have to write this traversal into the tar ourselves? 

This tar file deals with symlinks for example: https://github.com/coder/coder/blob/8d668963b915e598044c9ff349fd8ea8535405df/provisionersdk/archive.go#L45-L148

Maybe we can have a standard tool?

Reply:
eh. my implementation here is pretty short and straight forward. plus it has some details that may not translate to a generic implementation well like only capturing files inside .terraform/modules/ while also preserving that part of the path inside the archive.